nestjs/src/cross-check/cross-check.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Repository } from 'typeorm';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { HttpService } from '@nestjs/axios';
+import { Cron } from '@nestjs/schedule';
+import { ConfigService } from '../config';
+import { isTaskNeededToStart, isTaskNeededToFinish } from './utils/tasks-filtering';
+import { CourseTask } from '@entities/courseTask';
+import { from, catchError, mergeMap, EMPTY, toArray, lastValueFrom } from 'rxjs';
+
+@Injectable()
+export class CrossCheckService {
+  private readonly logger = new Logger('CrossCheckService');
+
+  constructor(
+    @InjectRepository(CourseTask)
+    private readonly courseTaskRepository: Repository<CourseTask>,
+    private readonly httpService: HttpService,
+    private readonly conf: ConfigService,
+  ) {}
+
+  @Cron('5 0 * * *', { timeZone: 'UTC' })
```

Comment:
Cron is tricky to read for newcomers. I would add comment or self descriptive variable to explain what `5 0 * * *` means